internal/client/client.go

```diff
@@ -453,6 +453,31 @@ func (c *CSAPI) MustDoFunc(t *testing.T, method string, paths []string, opts ...
 	return res
 }
 
+// MustDoFuncUntil is the same as MustDoFunc but retries up to 5 times, with 500 ms intervals, if the condition doesn't match.
+func (c *CSAPI) MustDoFuncUntil(t *testing.T, method string, paths []string, checkFunc func(*http.Response) bool, opts ...RequestOpt) *http.Response {
+	t.Helper()
+	var res *http.Response
+	res = nil
+	for retryNum := 0; retryNum < 5; retryNum++ {
+		res = c.DoFunc(t, method, paths, opts...)
+		if res.StatusCode < 200 || res.StatusCode >= 300 {
+			defer res.Body.Close()
+			body, _ := ioutil.ReadAll(res.Body)
+			t.Fatalf("CSAPI.MustDoFuncUntil response return non-2xx code: %s - body: %s", res.Status, string(body))
+			return res
+		}
+
+		success := checkFunc(res)
+		if t.Failed() || success {
+			return res
+		}
+
+		time.Sleep(500 * time.Millisecond)
```

Comment:
If the average race is resolved quicker than this, say 100ms, then this is a little pessimistic. Might exponential backoff (rather than linear) make more sense? (e.g. 10ms delay, quadrupling every retry)

Reply:
This is too inflexible. Some endpoints may take longer than others to update, it needs to be configurable how long to keep trying for.